src/components/molecules/AnnouncementMessage/AnnouncementMessage.tsx

```diff
@@ -1,53 +1,105 @@
+import { useShowHide } from "hooks/useShowHide";
 
 import { RenderMarkdown } from "components/organisms/RenderMarkdown";
+import { LinkButton } from "components/atoms/LinkButton";
 
 import "./AnnouncementMessage.scss";
 
-type AnnouncementMessageProps = {
-  message?: string;
-};
+export interface AnnouncementMessageProps {
+  banner: Banner;
+  announcementForUser?: boolean;
+}
 
 export const AnnouncementMessage: React.FC<AnnouncementMessageProps> = ({
-  message = "",
+  banner,
+  announcementForUser: isAnnouncementForUser = false,
 }) => {
-  const [isVisible, setVisibility] = useState<boolean>(false);
-  const { isExpanded } = useChatSidebarControls();
-
-  const hideAnnouncement = useCallback(() => {
-    setVisibility(false);
-  }, []);
+  const {
+    isShown: isAnnouncementMessageVisible,
+    show: showAnnouncementMessage,
+    hide: hideAnnouncementMessage,
+  } = useShowHide();
 
   useEffect(() => {
-    if (message) {
-      setVisibility(true);
+    if (isDefined(banner?.content)) {
+      showAnnouncementMessage();
     }
-  }, [message]);
+  }, [banner, showAnnouncementMessage]);
+
+  const isActiveButton =
+    banner?.buttonDisplayText && banner?.buttonUrl && banner?.isActionButton;
+
+  const isButtonShown = isActiveButton && banner.buttonUrl;
+  const isAnnouncementCloseable =
+    isAnnouncementForUser && !banner?.isForceFunnel;
+
+  const containerClasses = classNames("AnnouncementMessage__container", {
+    "AnnouncementMessage__container--centered": banner?.isFullScreen,
+    "AnnouncementMessage__container--admin": !isAnnouncementForUser,
+  });
+
+  const announcementMessageClasses = classNames("AnnouncementMessage", {
+    "AnnouncementMessage--fullscreen":
+      banner?.isFullScreen && isAnnouncementForUser,
+    "AnnouncementMessage--user": isAnnouncementForUser,
+  });
 
-  if (!isVisible || !message) return null;
+  const handleBannerModalClose = () => {
+    if (banner?.isForceFunnel) return;
+
+    hideAnnouncementMessage();
+  };
+
+  if (!isAnnouncementForUser && !banner?.content)
+    return (
+      <div className="AnnouncementMessage">
+        <span className="AnnouncementMessage__default-text">
+          No announcement
+        </span>
+      </div>
+    );
+
+  if (!isAnnouncementMessageVisible) return null;
 
   return (
-    <div
-      className={classNames("announcement-container", {
-        centered: !isExpanded,
-      })}
-    >
-      <div className="announcement-message">
-        <RenderMarkdown text={message} />
+    <>
+      <div className={containerClasses} onClick={handleBannerModalClose} />
+
+      <div className={announcementMessageClasses}>
+        {banner?.title && (
+          <h2 className="AnnouncementMessage__title">{banner.title}</h2>
+        )}
+
+        <div className="AnnouncementMessage__content">
+          <RenderMarkdown text={banner?.content} />
+        </div>
+
+        {isButtonShown && (
+          <LinkButton
+            href={banner.buttonUrl || ""}
```

Comment:
Maybe add this to the conditional check and not render it at all if this is missing?

Reply:
Definitely. Don't want to link to an empty string.

Reply:
`isButtonShown` alreadfy includes that, but VS Code highlights an error if I delete the `|| ""` here
so leaving it simply fixes the issue with vs code and not the actual logic

Reply:
You can just use them inline like this `{isActiveButton && banner.buttonUrl && (` and remove `|| ""`

The other way around this would be `href={banner.buttonUrl!}` but it's a tricky thing that can blow in our face in the future

Reply:
fixed 947713e